src/Tasks/ManifestUtil/ManifestReader.cs

```diff
@@ -269,3 +265,38 @@ public ComInfo(string manifestFileName, string componentFileName, string clsid,
         public string TlbId { get; }
     }
 }
+
+namespace System.IO
+{
+    public static class StreamExtensions
+    {
+        public static void ReadExactly(this Stream stream, byte[] buffer, int offset, int count)
+        {
+#if !NET
```

Comment:
nit: I'd wrap the whole class/namespace instead. On a quick read I thought this would cause an empty method to be called on NET, but this works because the compiler prefers the method on the class over the extension method in that case. Would be nice to make this more explicit.

Reply:
Hi @akoeplinger , I tried to wrap the whole class/namespace but failed. The code runs successfully on all other platforms, but fails on "Windows Full Release (no bootstrap)" with CP0001 error. During local testing, I observed that although the extension method wasn't being wrapped, the class implementation takes precedence in the compilation. Since "Windows Full Release (no bootstrap)" passed all tests successfully, I proceeded with the code modifications shown above.

Reply:
ah that's actually a good finding. CP0001 means
> CP0001 | A type, enum, record, or struct visible outside the assembly is missing in the compared assembly when required to be present.

that's because `StreamExtensions` is public but it shouldn't be.

Reply:
I considered changing this public class/method to internal, however, doing so would prevent files in other namespaces from accessing it:
https://github.com/dotnet/msbuild/blob/d8efe63fdc2c38a04d1fb1be7fc99d807453b664/src/Utilities.UnitTests/TrackedDependencies/FileTrackerTests.cs#L1738

I'm wondering if it's better to keep the method public or proceed with making it internal and provide another extension method for other files?

Reply:
you can make it internal and move this to a Util.cs somewhere that gets included in projects that need the method

Reply:
Update: https://github.com/dotnet/msbuild/pull/11626/commits/1a7e84839dde751f123c4d94e9749a6ce20d70cc